pkg/ds/farm_test.go

```diff
@@ -798,7 +792,42 @@ func TestMultipleFarms(t *testing.T) {
 	}
 
 	// Verify node2 is unscheduled
-	labeled, err = applicator.GetLabels(labels.POD, "node2/testPod")
-	Assert(t).IsNil(err, "Expected no error getting labels")
-	Assert(t).IsFalse(labeled.Labels.Has(DSIDLabel), "Expected pod not to have a dsID label")
+	labeled, err = waitForPodLabel(applicator, false, "node2/testPod")
+	Assert(t).IsNil(err, "Expected pod not to have a dsID label")
+}
+
+func waitForPodLabel(applicator labels.Applicator, hasDSIDLabel bool, podPath string) (labels.Labeled, error) {
+	var labeled labels.Labeled
+	var err error
+
+	condition := func() error {
+		var val bool
+		labeled, err = applicator.GetLabels(labels.POD, podPath)
+		if err == nil {
+			val = labeled.Labels.Has(DSIDLabel)
+			if hasDSIDLabel == val {
+				return nil
+			}
+		}
+		return util.Errorf("Expected hasIDLabel to be '%v', got '%v'", hasDSIDLabel, val)
+	}
+
+	err = waitForCondition(condition)
+	return labeled, err
+}
+
+func waitForCondition(condition func() error) error {
+	timeout := time.After(5 * time.Second)
```

Comment:
do we still see test flakes if this is set to 1 second? 0.5 seconds? This kinds of delays are very large compared to how quickly most of our test suite runs.